File: my-project-server/libs/Category.js
const mysql = require("mysql");

module.exports ={
    createCategory: async (pool, categoryName, imageUrl) => {
        var sql = "INSERT INTO category (category_name, imageUrl) "
                + "VALUES (?, ?) ";
        sql = mysql.format(sql, [categoryName, imageUrl]);

        return await pool.query(sql);
    },
    getAllCategory: async (pool) => {
        var sql = "SELECT * FROM "
                    + "category a "
    
        return await pool.query(sql);
    },

    getByCategoryId: async (pool, categoryId) => {
    var sql = "SELECT * FROM "
                + "category a "
                + "WHERE category_id = ?";

    sql = mysql.format(sql, [categoryId]);
    
    return await pool.query(sql);
},

updateCategory: async (pool, categoryId, categoryName, imageUrl ) => {
    var sql = `UPDATE category SET category_name = ?,  imageUrl = ? WHERE category_id = ?`;
    sql = mysql.format(sql, [categoryName, imageUrl, categoryId ]);

    return await pool.query(sql);
},
deleteCategory: async (pool, categoryId) => {
    var sql = "DELETE FROM category WHERE category_id = ?";
    sql = mysql.format(sql, [categoryId]);

    return await pool.query(sql);
}


}

File: my_project_app/lib/category.dart
class Category {
  final int categoryId;
  final String categoryName;
  final String imageUrl;

  const Category(
      {required this.categoryId,
      required this.categoryName,
      required this.imageUrl});

  factory Category.fromJSON(Map<String, dynamic> json) {
    return Category(
        categoryId: json["category_id"],
        categoryName: json["category_name"],
        imageUrl: json["imageUrl"]);
  }

  get isCircular => null;
}
